### PR TITLE
Android sense download  

### PR DESCRIPTION
**Pull request by menaka121**

### CODE REVIEW

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java [hunk 1]
```diff
@@ -206,6 +219,27 @@ public Device getDevice(
...
+        //TODO:: null check customDeviceName at UI level
+        try {
+
+            String sep = File.separator;
+            String sketchFolder = "repository" + sep + "resources" + sep + "sketches" + sep + "android_sense" + sep;
+            String archivesPath = CarbonUtils.getCarbonHome() + sep + sketchFolder;
+
+            Response.ResponseBuilder rb = Response.ok(new File(archivesPath+sep+"androidsense.apk"));
+            rb.header("Content-Disposition",
+                      "attachment; filename=\"" + "androidsense.apk" + "\"");
```
Comment (madhawap) [inline components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: Can you merge line number 234 and 235 to one line.

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java [hunk 2]
```diff
@@ -206,6 +219,27 @@ public Device getDevice(
 
     }
 
+    @Path("manager/device/{sketch_type}/download")
+    @GET
+    @Produces("application/octet-stream")
+    public Response downloadSketch(@PathParam("sketch_type") String sketchType) {
+        //TODO:: null check customDeviceName at UI level
+        try {
+
```
Comment (madhawap) [inline components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: Remove the new line here

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java [hunk 3]
```diff
@@ -206,6 +219,27 @@ public Device getDevice(
...
+            String archivesPath = CarbonUtils.getCarbonHome() + sep + sketchFolder;
+
+            Response.ResponseBuilder rb = Response.ok(new File(archivesPath+sep+"androidsense.apk"));
+            rb.header("Content-Disposition",
+                      "attachment; filename=\"" + "androidsense.apk" + "\"");
+            return rb.build();
+        } catch (IllegalArgumentException ex) {
+            return Response.status(400).entity(ex.getMessage()).build();//bad request
+        }
+
```
Comment (sinthuja) [inline components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: Remove unnecessary blank lines.

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/plugin/impl/AndroidSenseManagerService.java [hunk 4]
```diff
@@ -19,17 +19,17 @@ public String getType() {
 		return AndroidSenseConstants.DEVICE_TYPE;
 	}
 
-	@Override
+//	@Override
```
Comment (sinthuja) [inline components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/plugin/impl/AndroidSenseManagerService.java]: Remove this if it's not needed, don't comment it out.

Comment (menakaj) [reply components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/plugin/impl/AndroidSenseManagerService.java]: This class is used now as the structure of the device plugin is changed. Hence this class has been removed from the repository.

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java (hunk 3 above)
Comment (menakaj) [reply components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: This class is not used now as the structure of the device plugin is changed. Hence this class has been removed from the repository.

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java (hunk 2 above)
Comment (menakaj) [reply components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: This class is not used now as the structure of the device plugin is changed. Hence this class has been removed from the repository.

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java (hunk 3 above)
Comment (menakaj) [reply components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: This class is not used now as the structure of the device plugin is changed. Hence this class has been removed from the repository.

File: components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java (hunk 1 above)
Comment (menakaj) [reply components/device-mgt-iot-androidsense/org.wso2.carbon.device.mgt.iot.androidsense.api/src/main/java/org/wso2/carbon/device/mgt/iot/androidsense/service/impl/AndroidSenseService.java]: This class is not used now as the structure of the device plugin is changed. Hence this class has been removed from the repository.

**Pull request merged**